react/features/lobby/middleware.js

```diff
@@ -170,6 +141,62 @@ StateListenerRegistry.register(
     }
 );
 
+/**
+ * Function to handle the lobby notification.
+ *
+ * @param {Object} store - The Redux store.
+ * @returns {void}
+ */
+function _handleLobbyNotification(store) {
+    const { dispatch, getState } = store;
+    let notificationTitle;
+    let customActionNameKey;
+    let customActionHandler;
+    let descriptionKey;
+    let icon;
+
+    const knockingParticipants = getKnockingParticipants(getState());
+    const firstParticipant = knockingParticipants[0];
+
+    if (knockingParticipants.length > 1) {
+        descriptionKey = 'notify.participantsWantToJoin';
+        notificationTitle = i18n.t('notify.waitingParticipants', {
+            waitingParticipants: knockingParticipants.length
+        });
+        icon = NOTIFICATION_ICON.PARTICIPANTS;
+        customActionNameKey = [ 'notify.viewLobby' ];
+        customActionHandler = [ () => batch(() => {
+            dispatch(hideNotification(LOBBY_NOTIFICATION_ID));
+            dispatch(openParticipantsPane());
+        }) ];
+    } else {
+        descriptionKey = 'notify.participantWantsToJoin';
+        notificationTitle = firstParticipant.name;
+        icon = NOTIFICATION_ICON.PARTICIPANT;
+        customActionNameKey = [ 'lobby.admit', 'lobby.reject' ];
+        customActionHandler = [ () => batch(() => {
+            dispatch(hideNotification(LOBBY_NOTIFICATION_ID));
+            dispatch(approveKnockingParticipant(firstParticipant.id));
+        }),
+        () => batch(() => {
+            dispatch(hideNotification(LOBBY_NOTIFICATION_ID));
+            dispatch(rejectKnockingParticipant(firstParticipant.id));
+        }) ];
+    }
+    if (knockingParticipants.length === 0) {
```

Comment:
Can you please move this up? Otherwise the code will first create a notification that will never be used. We can check this the first thing and return early.